### PR TITLE
feat: add ability to enter API keys in the UI

### PR DESCRIPTION
**Pull request by ali00209**
## Summary
Implements UI components and functionality to allow users to securely input, save, and manage API keys through the application interface.
## UI Screenshot
image

### CODE REVIEW

**Review by karrot0 — APPROVED** [review]

**Comment by wonderwhy-er** [comment]
Want to add this too asap, @ali00209 why is it a draft? I see karrot0 approved, we could merge

**Comment by chrismahoney** [comment]
Taking a look at this shortly

**Comment by emcconnell** [comment]
@ali00209 I pulled this locally and merged it with latest so I could test. It looks good to me. I tested with OpenRouter, Anthropic, and OpenAI. Do you want to resolve the conflict and we should be able to merge in? Thanks!

**Comment by chrismahoney** [comment]
This is similar to #124, but more of a simple step toward the Settings UI. Thanks for testing it @emcconnell, if we're in agreement let's push this in as a first step to surface provider and model-specific settings? I'm cool with that.

File: app/utils/constants.ts [hunk 1]
```diff
@@ -4,17 +4,17 @@ export const WORK_DIR_NAME = 'project';
 export const WORK_DIR = `/home/${WORK_DIR_NAME}`;
 export const MODIFICATIONS_TAG_NAME = 'bolt_file_modifications';
 export const MODEL_REGEX = /^\[Model: (.*?)\]\n\n/;
-export const DEFAULT_MODEL = 'claude-3-5-sonnet-20240620';
```
Comment (chrismahoney) [inline app/utils/constants.ts:7]: Oof, we don't want to change the default model. I'll open up a ticket for this.

**Review by chrismahoney — APPROVED** [review]
I'm just going to open an issue to revert the default model until a better default system is in place, but we should let this in to start our review and approval process.

**Comment by chrismahoney** [comment]
Or not, will re-review after the conflicts are resolved.

**Comment by chrismahoney** [comment]
I'm merging this for now, but opening an issue for the pnpm failures during github action runs tomorrow. Entering an API key for Ollama and OpenAILike behaviors worked for me cleanly.

**Pull request merged**